### PR TITLE
Daml-LF: make Numeric scale type safe

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
This PR advances the Numeric implementation #2289.

This PR makes the numeric scale type safe: It can only take value between 0 and maxPrecision.
This will simplifiy future changes, and avoid unnecessary dynamic check of bounds.

### Pull Request Checklist

- [X] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [X] Include appropriate tests
- [X] Set a descriptive title and thorough description
- [X] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [X] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/NumericModule.scala [hunk 1]
```diff
@@ -35,6 +35,15 @@ abstract class NumericModule {
     */
   val maxPrecision = 38
```
Comment (associahedron) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/NumericModule.scala]: As discussed, max scale should be 37

```suggestion
   val maxPrecision = 37
```

Comment (remyhaemmerle-da) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/NumericModule.scala]: I will do it in a subsequent PR.

Comment (associahedron) [reply daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/NumericModule.scala]: ^ ah, I misunderstood. Yes, that's right.

File: daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/NumericModule.scala [hunk 2]
```diff
@@ -242,10 +250,11 @@ abstract class NumericModule {
     *
     * With use this function to convert BigDecimal with unknown scale.
```
Comment (gerolf-da) [inline daml-lf/data/src/main/scala/com/digitalasset/daml/lf/data/NumericModule.scala]:
```suggestion
    * Use this function to convert BigDecimal with unknown scale.
```

**Pull request merged**